packages/webdriver/src/local.ts

```diff
@@ -0,0 +1,110 @@
+import 'chromedriver';
+import 'geckodriver';
+
+import { Operation, resource } from 'effection';
+import { spawn as spawnChildProcess } from '@effection/node';
+import { Atom } from '@bigtest/atom';
+import { once } from '@effection/events';
+
+import { fetch } from './fetch';
+import { findAvailablePortNumber } from './find-available-port-number';
+import { untilURLAvailable } from './until-url-available';
+import { WebDriver, Options } from './interfaces';
+
+type LocalDriverName = 'chromedriver' | 'geckodriver';
+
+/**
+ * Create a local `WebDriver` resource based on `driverName` (either 'geckodriver' or
+ * 'chromedriver').
+ *
+ * In order to function, the local executables for `Chrome` and
+ * `Firefox` must be separately installed onto the machine. So in CI
+ * for example, you would need to make sure that the browsers are
+ * present on your container.
+ *
+ * In order to keep things simple, and because the geckodriver only
+ * suopports a single Firefox session per driver process, we spawn a
+ * new process for each local driver.
+ */
+export function * Local(driverName: LocalDriverName, options: Options): Operation<WebDriver> {
+
+  let port: number = yield findAvailablePortNumber();
+  let driverURL = `http://localhost:${port}`;
+
+  let session: { sessionId: string };
+
+  let capabilities = new Atom(Capabilities);
+
+  if (options.headless) {
+    capabilities.slice<string[]>(['alwaysMatch', 'goog:chromeOptions', 'args'])
+      .over(args => args.concat(['--headless']))
+    capabilities.slice<string[]>(['alwaysMatch', 'moz:firefoxOptions', 'args'])
+      .over(args => args.concat(['--headless']))
+  }
+
+  let driver: WebDriver = {
+    *navigateTo(url: string): Operation<void> {
+      let response = yield fetch(`${driverURL}/session/${session.sessionId}/url`, {
+        method: 'post',
+        body: JSON.stringify({ url })
+      });
+
+      if (!response.ok) {
+        throw new Error(`GetError: GET ${url}\n${response.status} ${response.statusText}`);
+      }
+    }
+  };
```

Comment:
How do you feel about making this a class instead?

Reply:
That would probably be wise. Looking at this code, it seems like there's actually nothing "local" about it. It's independent of the fact that the Webdriver server was started by running a local executable.

Reply:
done.